client/state/wp-data/calypso-registry-plugin.js

```diff
@@ -0,0 +1,92 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { mapValues, without } from 'lodash';
+
+const calypsoRegistryPlugin = calypsoStore => registry => {
+	const namespaces = {};
+	let listeners = [];
+	let lastCalypsoState = calypsoStore.getState();
+
+	function calypsoListener() {
+		listeners.forEach( listener => listener() );
+	}
+
+	calypsoStore.subscribe( () => {
+		const calypsoState = calypsoStore.getState();
+		const hasChanged = calypsoState !== lastCalypsoState;
+		lastCalypsoState = calypsoState;
```

Comment:
This assigment could also go into the `if` clause below, no?

Reply:
Yes, that would be better, thanks!